modules/api_docs/php/api_docs.class.inc

```diff
@@ -0,0 +1,105 @@
+<?php declare(strict_types=1);
+/**
+ * PHP Version 7
+ *
+ * @category Documentation
+ * @package  Main
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+namespace LORIS\api_docs;
+
+/**
+ * This class list the schema files of this LORIS instance modules
+ * for the swagger-ui app to present. It is a public module but will only
+ * show specification (schemas) of module that the user can access.
+ *
+ * @category Documentation
+ * @package  Main
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+class API_Docs extends \NDB_Page
+{
+    /**
+     * Determine whether the user has permission to view this page
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool whether the user has access
+     */
+    public function hasAccess(\User $user): bool
+    {
+        return true;
+    }
+
+    /**
+     * Sets the template variable for the schema file to load.
+     *
+     * @return void
+     */
+    function setup()
+    {
+        $baseurl = \NDB_Factory::singleton()
+            ->config()
+            ->getSetting('base');
```

Comment:
The BaseURL should come from `\NDB_Factory::singleton()->settings()->getBaseURL()` when required.

Reply:
now using `$baseurl = $request->getAttribute('baseurl');` instead